x-pack/plugins/fleet/server/plugin.ts

```diff
@@ -206,6 +206,10 @@ export class FleetPlugin
     this.encryptedSavedObjectsSetup = deps.encryptedSavedObjects;
     this.cloud = deps.cloud;
 
+    const config = await this.config$.pipe(first()).toPromise();
+
+    appContextService.fleetServerEnabled = config.agents.fleetServerEnabled;
```

Comment:
Open to going a different route here 🙂

Reply:
I honestly view `appContextService` as "here, hold this for me" and otherwise meaningless. So, sure.

Reply:
I think we can do it like this `appContextService.getConfig().?agents?.fleetServerEnabled` so we do not have to add another property on the appContext service

Reply:
@nchaulet did not realize that `appContextService` had methods to grab the config - I should have looked harder 😄 .
that being said, I can't use `appContextService.getConfig()` (or `appContextService.getConfig$()`) because it will only be populated after calling `appContextService.start()` which is done in the Plugin's `start()` phase and not in `.setup()`.

Reply:
ah I see in this case I would probably add the config as an argument of `registerSavedObject` and pass it through the migration